Add citation information, and so much more

brinkdp opened the pull request:
Citation information will need to be implemented for several different aspects of the genome portal. For instance: for the service itself (EB-132 & EB-173), for the code (EB-189), for the metadata records (EB-191), and for JBrowse 2 (EB-202). 

Made this into to a draft PR for now, but we can consider splitting the work into several PRs if that is more convenient.

--- 
The initial commit (41f25ba) of the PR is about citation of the code, and as such relates to the upcoming refinement of README.md (EB-188; @kwentine), and Zenodo integration (EB-174). The commit explores `CITATION.cff`, which is used to add authorship to the repository, and to pass metadata to Zenodo once publishing of `genome-portal` releases have been integrated. Author names are currently populated with placeholders since authorship details are yet to be finalized.

With `CITATION.cff` in place, a `Cite this repository` option can be found in the **About** menu in the top right corner of the GitHub repo upon navigating to this branch. Since this worked as expected, I suggest that we proceed to add the following standard statement used by e.g. Data Platform and Pathogens Portal to `README.md` (did not commit this yet to avoid conflict with #54): 

```
## Cite this code

Click on 'Cite this repository' near the top right of this repository to see how to formally cite this repository.
```

Another feature of `CITATION.cff` I explored was the option to include a reference to JBrowse 2. This was inspired by EB-202 and by reading the [documentation for the .cff format](https://citation-file-format.github.io/) and the [.cff schema](https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md). The idea is to treat the `references` array in `CITATION.cff` analogously to the bibliography section in a scientific publication. It does bloat the `.cff `file, but I think the result is useful. However, I am curious to learn exactly how this would be integrated in the Zenodo entry metadata.

RMCrean commented on hugo/content/citation/_index.md:
```diff
@@ -0,0 +1,91 @@
+
+Below are instructions for how to cite different aspects of the Genome Portal. These include:
+
+- Citing the the Swedish Reference Genome Portal website
+
+- Citing a specific species entry in the Genome Portal
+
+- Citing the data in the Genome Portal
+
+- Citing the underlying code of the Genome Portal, including its dependency on JBrowse 2
```
Optional, but this could be written as a table of contents with clickable links to each section below.

brinkdp replied:
Good idea! Added to board as EB-225.

RMCrean reviewed (commented):
One suggestion which can be ignored. 

Not marking approved as I will now work on this branch (as planned) to add the user guide and this page to the navbar.

kwentine commented on CITATION.cff:
```diff
@@ -0,0 +1,82 @@
+license: MIT
+references:
+  - authors:
+      - family-names: Diesh
```
I wonder if inserting the whole author list is necessary 🤔 According to JBrowse2 [citations instructions](https://jbrowse.org/jb2/), article and DOI should be satisfactory.

kwentine commented on hugo/content/citation/_index.md:
```diff
@@ -0,0 +1,91 @@
+It aims to promote the identification, discovery, and reuse of research resources. Research Resource Identifiers (**RRIDs**) are persistent and unique identifiers for
+referencing a research resource. For official guidance see the [SciCrunch page on RRID citations](https://scicrunch.org/resources/about/guidelines).
+
+The RRID for the Swedish Reference Genome Portal is [**XXXXXXXX**](https://XXXXXXXXXX).
```
I would maybe put the general information about PRIDs (that I do find very valuable) as a hover-popup or footnote, referred to by the first PRID reference on this line.

kwentine commented on hugo/content/citation/_index.md:
```diff
@@ -0,0 +1,91 @@
+**Reference list (APA format)**:\
+Swedish Reference Genome Portal (_access date_), SciLifeLab Data Centre. Version (version number). <https://genomes.scilifelab.se>, RRID:XXXXXXXX. [Species DOI].
+
+You will find the version number of the portal at XXXXXthe bottom of the footer on any pageXXXXXX, or on our
```
Accessing a past version of the portal may be tricky: the party wishing to verify the citation would need to checkout the code at the corresponding version tag, build the data, and run hugo themselves. But it is in principle possible, if our upstream data sources don't change or vanish, that is.

kwentine commented on hugo/content/citation/_index.md:
```diff
@@ -0,0 +1,91 @@
+### 2. Citing a specific species entry
+
+If you would like to refer to a specific species entry from the Genome Portal, you can do so by taking the Genome Portal citation (Section 1) and adding the species name and the
+DOI pointing to its associated metadata in SciLifeLab Data Repository:
```
Should we provide a link to the data repository here perhaps ?

brinkdp replied:
Sure!

kwentine commented on hugo/content/citation/_index.md:
```diff
@@ -0,0 +1,91 @@
+
+where `[species_name]` and `[DOI]` should be replaced as described above.
+
+The Genome Portal staff can help you reserve an entry on the portal and a DOI for the metadata already during the manuscript submission stage. When the manuscript is accepted for publication and the data has been
```
This part could possibly be moved to the "Contribute" section

brinkdp replied:
I like the idea. It would reduce the amount of text and hopefully make the citation page slightly more readable. I drafted a little something in [46b7e1f](https://github.com/ScilifelabDataCentre/genome-portal/pull/55/commits/46b7e1f4d7f94a6b570e5f52a868a6d3e3d5500e).

kwentine commented on hugo/content/citation/_index.md:
```diff
@@ -0,0 +1,91 @@
+we reccomend that you cite the original publication. For datasets that do not have an associated publication, please go to the _Download_ page for the species, click the *Show reduced table view*
+button and see if there is an accession number or a DOI that can be used to refer to the dataset.
+
+### 4. Citing the underlying code
```
```suggestion
### 4. Citing the source code
```

kwentine commented on hugo/content/citation/_index.md:
```diff
@@ -0,0 +1,91 @@
+
+#### Citing the Genome Portal code
+
+The Genome Portal is operated by the <a target="_blank" href="https://scilifelab.se/data">SciLifeLab Data Centre</a> and partners. All of the source code used on the website is available on GitHub.
```
```suggestion
The Genome Portal is operated by the <a target="_blank" href="https://scilifelab.se/data">SciLifeLab Data Centre</a> and partners. All of the source code used to generate the website is available on [GitHub](https://github.com/ScilifelabDataCentre/genome-portal/).
```

kwentine reviewed (commented):
What a work you did 🙌🏻 ! I learnt a lot about citation practices by doing the review. To be honest I felt a bit overwhelmed by all the instructions. Maybe a compilation of all the citation format (a digest) could be given at the outset (in an info box or something), followed by the more detailed explanations ?

brinkdp commented:
I think this PR is ready for another round of review! I've reworked text after the Workday ideas, aiming to slim down the text so that the genome portal citation is only declared once (in the `#citing-the-website-and-the-source-code` section). 

Then, based on feedback from Johan R, the exact phrasing of the citation was altered, and the source code citation block was added back in. 

Commit 9f0e24d contains the new suggestion that I think contains everything we need before launch. The species description pages have been updated with citation information too. Could you please take an extra careful look at the phrasing on those pages (since it is new content)?

Another thing I noticed is that we were suggested to put https://www.genomes.scilifelab.se/ in the citation, like Pathogens do. However, the w3 version does not resolve, so I've changed it in the citation to https://genomes.scilifelab.se/ for now. I guess it would be good for us to enable https://www.genomes.scilifelab.se/ too?

Minor comments:
- The Zenodo DOI part of Cite Us is temporarily removed awaiting the actual DOI.
- A TODO is to add a hyperlink from SCR_026008 to the RRID summary page when it is live. 
- The navbar commit includes the User guide link. This page is not populated yet, so maybe we want to drop it from the navbar until the content is ready?

apfuentes commented:
Really good job @brinkdp :clap:! The text reads very well and the sections are better defined. There is probably still room to reduce the word count, but that could be done in a separate PR.

I added a few comments on the code, but I couldn't figure out how to add comments to the "not-compared" sections, so I am adding them here:

- The text using H4 [#### Citing the website and the source code] is hard to distinguish from bold normal text, so I suggest to use H2 ## and H3 ### instead, and across the page. If H2 is too big, then we might need to set an intermediate size in css.

- In the [Citing the website and the source code] section, the in-text citation for the genome portal is missing

- We should unify terms: Species entry vs. Species page, I have used the latter in the User guide.

- If we decide to use the terms in the User guide, then replace "Citing a specific species entry" for "Citing a Species page"

- If we decide to use the terms in the User guide, then replace "Description page" for "Description tab"

- In [the binomal name of your species in italics], replace "your" for "the"

- Replace "Data availability statement for submitting researchers" for "In-text citation in the Data Availability Statement of a manuscript"

- In [under the MIT license], replace "the" for "a"

- In [Citing the source code], why not to spell out the Zenodo citation there instead of directing users to visit github? As it was done in the Pathogens portal https://www.pathogens.se/citation/

- If we decide to use the terms in the User guide, in [to cite each dataset is found on the _Description_ page of each species] replace "page" for "tab"


- For clarity, perhaps the sections could be reordered in this way:

1. Citing the website

- Citing the Genome Portal website
In-text citation
Reference

- Citing a Species page in the Genome Portal website
In-text citation
Reference
In-text citation for the Data Availability Statement of a manuscript

2. Citing the source code of the website
In-text citation
Reference

3. Citing the genomic data in a Species page

4. Citing the JBrowse 2 genome browser

Let me know if you wanna have a chat about this in person :relaxed:

apfuentes reviewed (commented):
Really good job @brinkdp 👏! The text reads very well and the sections are better defined. There is probably still room to reduce the word count, but that could be done in a separate PR.

I added a few comments on the code, but I couldn't figure out how to add comments to the "not-compared" sections, so I am adding them here:

- The text using H4 [#### Citing the website and the source code] is hard to distinguish from bold normal text, so I suggest to use H2 ## and H3 ### instead, and across the page. If H2 is too big, then we might need to set an intermediate size in css.

- In the [Citing the website and the source code] section, the in-text citation for the genome portal is missing

- We should unify terms: Species entry vs. Species page, I have used the latter in the User guide.

- If we decide to use the terms in the User guide, then replace "Citing a specific species entry" for "Citing a Species page"

- If we decide to use the terms in the User guide, then replace "Description page" for "Description tab"

- In [the binomal name of your species in italics], replace "your" for "the"

- Replace "Data availability statement for submitting researchers" for "In-text citation in the Data Availability Statement of a manuscript"

- In [under the MIT license], replace "the" for "a"

- In [Citing the source code], why not to spell out the Zenodo citation there instead of directing users to visit github? As it was done in the Pathogens portal https://www.pathogens.se/citation/

- If we decide to use the terms in the User guide, in [to cite each dataset is found on the Description page of each species] replace "page" for "tab"

For clarity, perhaps the sections could be reordered in this way:

1. Citing the website

1.1 Citing the Genome Portal website
In-text citation
Reference

1.2 Citing a Species page in the Genome Portal website
In-text citation
Reference
In-text citation for the Data Availability Statement of a manuscript

2. Citing the source code of the website
In-text citation
Reference

3. Citing the genomic data in a Species page

4. Citing the JBrowse 2 genome browser

Let me know if you wanna chat about this in person ☺️

brinkdp commented:
Thanks @apfuentes! 

>     * The text using H4 [#### Citing the website and the source code] is hard to distinguish from bold normal text, so I suggest to use H2 ## and H3 ### instead, and across the page. If H2 is too big, then we might need to set an intermediate size in css.

Good point. I personally think that h2 is too big 😄, but my bias comes from how it looks on my computer setup so we should look into this. I think that the reordering of sections that you proposed helped differentiate headings from bold: each heading in the latest commit now starts with a positional number. The bolded sections do not. 

Nevertheless, I think it would be good to harmonizing heading sizes across the pages. Perhaps even working on the css. I've make task for that (EB-272).

>     * In the [Citing the website and the source code] section, the in-text citation for the genome portal is missing

Indeed. This was done on purpose to reduce the complexity and word count. I assumed that it would be implied that a user would enter this in their reference manager software and add the citation in-text as usual. But if that is not clear, we can of course add the in-text example back in!

>     * We should unify terms: Species entry vs. Species page, I have used the latter in the User guide.

Yes, good idea! So the options are: A) Species page with tabs (description tab, assembly tab, download tab); and B) Species entry with description page, assembly page, and Species entry with pages (description page, assembly page, download page). Entry sounds more formal to me, but Perhaps A is better. I have implemented A in the new commit.

But do we call it a Genome Browser tab or page? It would be easier to go with tab if we decide on A, but it does not feel like a tab to me 🤔  

> 
>     * In [the binomal name of your species in italics], replace "your" for "the"
> 

Nicely spotted, this is lab slang. Fixed!

>     * Replace "Data availability statement for submitting researchers" for "In-text citation in the Data Availability Statement of a manuscript"
> 
>     * In [under the MIT license], replace "the" for "an"
> 
👍 Fixed! 

>     * In [Citing the source code], why not to spell out the Zenodo citation there instead of directing users to visit github? As it was done in the Pathogens portal https://www.pathogens.se/citation/
> 

Ah, yes, that is indeed the plan once we get the DOI! I had dropped it for now (9f0e24d) so that this page could be included for the launch, even if the we do not have the link to the DOI on the live site by then. The sentence directing users to GitHub was a temp fix to make the page self-contained without the Zenodo DOI. I have added the Zenodo placeholder back in now.

> 
> For clarity, perhaps the sections could be reordered
> 
 
Nice, I think numbering the subheadings helps. I've implemented a variant of what you suggestions, but am happy to keep discussing this!

RMCrean reviewed (approved):
Great job, and nice work fixing the conflict :). Some minor spelling changes, otherwise looks good. 

> The navbar commit includes the User guide link. This page is not populated yet, so maybe we want to drop it from the navbar until the content is ready?

We can drop it, but once Angela's PR is merged it will be populated. So as long as we don't make V1.0.0 before that I think it is fine to keep? - but we should make sure that has happened and otherwise hide/remove it. Does that sound good? 

> Another thing I noticed is that we were suggested to put https://www.genomes.scilifelab.se/ in the citation, like Pathogens do. However, the w3 version does not resolve, so I've changed it in the citation to https://genomes.scilifelab.se/ for now. I guess it would be good for us to enable https://www.genomes.scilifelab.se/ too?

Good point, we can work on making this happen. It seems like we just need to add an extra line to the kubernetes config: https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#redirect-fromto-www  I'll send Quentin a slack about it now as I think he is working on these files ATM.

brinkdp commented:
Thanks for that!

> We can drop it, but once Angela's PR is merged it will be populated. So as long as we don't make V1.0.0 before that I think it is fine to keep? - but we should make sure that has happened and otherwise hide/remove it. Does that sound good?
>
Sounds like a plan!

kwentine commented on hugo/content/species/meganyctiphanes_norvegica/_index.md:
```diff
@@ -35,12 +35,16 @@ The *Meganyctiphanes norvegica* species of krill is a macrozooplankton crustacea
 
 - <p> Unneberg, P., Larsson, M., Olsson, A., Wallerman, O., Petri, A., Bunikis, I., Vinnere Pettersson, O., Papetti, C., Gislason, A., Glenner, H., Cartes, J. E., Blanco-Bercial, L., Eriksen, E., Meyer, B., & Wallberg, A. (2024). Ecological genomics in the Northern krill uncovers loci for local adaptation across ocean basins. Nature Communications, 15(1), 6297. <a href="https://doi.org/10.1038/s4 ...
 
+If you have used the pages for this species in the Genome Portal, please refer to it in-text as: "The *Meganyctiphanes norvegica* entry in the Swedish Reference Genome Portal (access date)." and use the following for the bibliography:
```
In a later iteration, I would suggest moving this snippet to the template to avoid duplication. We could also thereby include the current version and date (but that may be double edged)

brinkdp replied:
Right! I put an (now outdated) version of this in the template in [b44928c](https://github.com/ScilifelabDataCentre/genome-portal/pull/55/commits/b44928c94546fd7595e6c610a83c824a48d96e94). But an update is due, thanks for reminding me.

kwentine replied:
I was ambiguous sorry: I meat in the HTML template where we could profit from `{{ $siteVersion }}` and `{{ .Date }}` type of things

brinkdp replied:
Ah yes, I like the idea. I thought of something similar myself. This could perhaps be related to the future investigation in EB-268 about hosting .bib/.ris citation files to improve UX.
